packages/lesswrong/server/notificationCallbacks.tsx

```diff
@@ -458,25 +458,31 @@ getCollectionHooks("Posts").newAsync.add(async function PostsNewNotifyUsersShare
   await createNotifications({userIds, notificationType: "postSharedWithUser", documentType: "post", documentId: _id})
 });
 
-getCollectionHooks("Posts").newAsync.add(async function CoauthorRequestNotifications(post: DbPost) {
-  const { _id, coauthorStatuses = [], hasCoauthorPermission } = post;
+getCollectionHooks("Posts").newAsync.add(async function CreateCoauthorRequestNotifications(post: DbPost) {
+  await sendCoauthorRequestNotifications(post);
+});
 
-  if (hasCoauthorPermission) {
-    return;
-  }
+getCollectionHooks("Posts").updateAfter.add(function UpdateCoauthorRequestNotifications(post: DbPost) {
+  return sendCoauthorRequestNotifications(post);
+});
 
-  const userIds = coauthorStatuses.filter(({ requested }) => !requested).map(({ userId }) => userId);
-  await createNotifications({userIds, notificationType: "coauthorRequestNotification", documentType: "post", documentId: _id});
+const sendCoauthorRequestNotifications = async (post: DbPost) => {
+  const { _id, coauthorStatuses, hasCoauthorPermission } = post;
 
-  await updateMutator({
-    collection: Posts,
-    documentId: _id,
-    data: {
-      coauthorStatuses: coauthorStatuses.map((status) => ({ ...status, requested: true })),
-    },
-    validate: false,
-  })
-});
+  if (hasCoauthorPermission === false && coauthorStatuses?.length) {
```

Comment:
I didn't understand until just know what you meant in the PR description about how checking for falsey hasCoauthorPermission was causing the bug.